### PR TITLE
Enable Node 16 and NPM 7/8 support on generation

### LINKED ISSUE #11135: Support node version 16.x
### Please describe your feature request

- [ ] **I have created my request on the Product Board before I submitted this issue**
- [ ] **I have looked at all the other requests on the Product Board before I submitted this issue**

### Summary

You should be able to build a Strapi application with node v16.x

### Why is it needed?

Node v16.x is going to become the new LTS release later this month.

### Suggested solution(s)

Update node version range in package.json to allow installing dependencies using node v16.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request.

### PR DESCRIPTION
**Pull request by derrickmehaffy**
### What does it do?

Modifies the new project generator to properly set the engines object to support Node 16

### Why is it needed?

Prepare for stable release

### How to test it?

- Generate a fresh project and confirm the root project package.json has the proper engine keys to match all other core packages
- Try to install node modules using both npm and yarn
- Try to start Strapi using Node 16

### Related issue(s)/PR(s)

fixes #11135

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
I have tested the beta.7 on node 16 and am not seeing any issues but I'll let e2e tests have a run at it.

**Review by alexandrebodin — APPROVED** [review]

**Comment by IamBlueSlime** [comment]
As everyone is moving their codebase to new Node LTS v16, could this PR be part of a v3 release too?
It seems complicated to wait for the v4 to release GA.
Thanks!

**Comment by derrickmehaffy** [comment]
> As everyone is moving their codebase to new Node LTS v16, could this PR be part of a v3 release too?
> It seems complicated to wait for the v4 to release GA.
> Thanks!

No, there are other breaking changes in v3 that make it impossible to add node 16 support to v3 without refactoring so we won't be backporting node 16 support to v3.

**Pull request merged**
